<launch>
   <group ns="$(env HOSTNAME)">

   <!-- Initialize the Onboard Setpoint Publisher -->
	<node pkg="bridge_px4" type="setpoint_publisher" name ="setpoint_publisher" output="screen">
	   <param name="sp_out_hz" value="200.0"/>
      <param name="checkup_hz" value="200.0"/>
      <param name="pos_hz_min" value="5.0"/>
      <param name="vel_hz_min" value="5.0"/>
      <param name="att_hz_min" value="100.0"/>
      <param name="checkup_hz_min" value="5.0"/>
      <param name="dt_fs" value="3.0"/>
      <param name="dt_rs" value="1.0"/>
	</node>

   <!-- Initialize the High Rate Controller 
   <node pkg="bridge_px4" type="hr_control" name ="hr_control" output="screen">
      <param name="pxy_slim" value="999.0"/>
      <param name="pz_slim" value="999.5"/>
      <param name="v_slim" value="999.5"/>
      <param name="q_slim" value="999.5"/>
      <param name="ep_lim" value="999.0"/>
      <param name="eq_lim" value="999.5"/>
   </node>
   -->

   <!-- Initialize MPC -->
   <node pkg="bridge_px4" type="MPC" name ="MPC" output="screen">
      <param name="x_box_lim" value="3.0"/>
      <param name="y_box_lim" value="1.75"/>
      <param name="z_box_lim" value="1.0"/>
      <param name="mass" value="0.53"/>
      <param name="setpoint_mode" value="1"/> <!-- 1: read from .csv, 2: circle, 3: lissajou -->
      <param name="s_sqrt_h" value="0.1"/>
      <param name="s_sqrt_v" value="0.1"/>
      <param name="r_sqrt_h" value="0.6"/>
      <param name="r_sqrt_v" value="0.6"/>
   </node>

   <!-- Initialize the Attitude Controller -->
   <node pkg="bridge_px4" type="att_control" name ="att_control" output="screen">
      <param name="max_thrust" value="14.81125"/>
   </node>

   <!-- Initialize the mavros bridge over serial -->
	<include file="$(find mavros)/launch/node.launch">
	   <arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
      <arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

	   <arg name="fcu_url" value="/dev/ttyS4:921600" />
      <arg name="gcs_url" value="" />
      <arg name="tgt_system" value="1" />
      <arg name="tgt_component" value="1" />
      <arg name="log_output" value="screen" />
      <arg name="fcu_protocol" value="v2.0" />
      <arg name="respawn_mavros" default="false" />
	</include>

   </group>

</launch>
